python/python/lancedb/query.py

```
@@ -544,6 +554,40 @@ def where(self, where: str, prefilter: bool = False) -> LanceVectorQueryBuilder:
         self._prefilter = prefilter
         return self
 
+    def rerank(
+        self, reranker: Reranker, query: Optional[str] = None
+    ) -> LanceVectorQueryBuilder:
+        """Rerank the results using the specified reranker.
+
+        Parameters
+        ----------
+        reranker: Reranker
+            The reranker to use.
+
+        query: Optional[str]
+            The query to use for reranking. This needs to be specified explicitly here
+            as the query used for vector search may already be vectorized and the
+            reranker requires a string query.
+            This is only required if the query used for vector search is not a string.
+            Note: This doesn't yet support the case where the query is multimodal or a
```

Can you just catch this and raise an exception in the unsupported case?